Gemfile

```diff
@@ -42,9 +46,10 @@ group :development do
 end
 
 group :test do
-  gem "capybara", ">= 2.15"
   gem "selenium-webdriver"
   gem "chromedriver-helper"
+  gem "shoulda-matchers", "~> 3.1"
```

Comment:
gem này dùng để làm gì?

Reply:
gem này giúp  viết test dễ dàng hơn, tiết kiệm được thời gian khi viết các test dài và phức tạp, code ngắn gọn, dễ đọc.

Reply:
- cung cấp cho ta nhiều phương thức dùng để test
  - have_secure_password
  - validate_length_of
  - validate_presence_of
và nhiều phương thức khác.
- Bình thường khi ta viết 1 đoạn test xem name có tồn tại hay không ta cần viết:
it "is invalid without a name" do
    contact = Contact.create(name: nil, email: "tungalm@gmail.com", password: "password")

    contact.valid?

    expect(contact.errors[:name]).to include("can't be blank")
  end
nhưng khi ta sử dụng gem này ta chỉ cần sử dụng phương thức validate_presence_of:
 it { is_expected.to validate_presence_of(:firstname) }
vì vậy nó giúp code của ta ngắn gọn và dễ đọc hơn